src/App.tsx

```diff
@@ -17,18 +72,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedTodos={selectedTodos}
+                handleSelectedTodos={handleSelectedTodos}
+                searchQuery={searchQuery}
+                handleSearchQuery={handleSearchQuery}
+                setSearchQuery={setSearchQuery}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {isFetching
+              ? <Loader />
+              : (
+                <div className="block">
+                  <TodoList
+                    visibleTodos={visibleTodos}
+                    handleClickModalButton={handleClickModalButton}
+                    isModalOpen={isModalOpen}
+                    personalTodo={personalTodo}
+                  />
+                </div>
+              )}
+
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {personalTodo
+        && isModalOpen
+        && (
+          <TodoModal
+            personalTodo={personalTodo}
+            setIsModalOpen={setIsModalOpen}
+          />
+        )}
```

Comment:
This thing looks way too complicated and as I understand because of it your modal *blinks* when you click the show button

If you look carefully, you may see this *blink* which happens before Loader appears

Reply:
good point! I changed this moment and now all is ok)